### PR TITLE
Replaces Batform with Vampire houses

### PR DESCRIPTION
**Pull request by tralezab**
## Permissions for species change

image

## About The Pull Request

Goodbye batform

Hello, new preference called vampire status. Outcast vampires act just like normal, Inoculated vampires join their department under a unified "vampire house name" everyone shares.

image

image

image

## Why It's Good For The Game

I've heard people complain about batform for years now, if vampires aren't gone by next halloween we can at least enjoy making them far less griefy (ruining a lot of the fun of halloween as just a dumb grief holiday) and more roleplay oriented. I don't even know why vampires got a griefy spell when they already have their main mechanic encourage randomly attacking people and stealing their blood to stay alive

## Changelog

:cl:
del: Batform is gone!
add: ...Replaced by vampire houses as a preference. Join your department as a vampire ménage!
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
bro this is literally just a roundstart message and a name change

**Comment by D4C-420** [comment]
Single issue I have with this is that this also affects vampires outside of Halloween. Vampires made by pride mirror or other methods should still retain their bat form on the basis that they would likely be the only vampires and they are a  non-round start race.

**Comment by tralezab** [comment]
> Single issue I have with this is that this also affects vampires outside of Halloween. Vampires made by pride mirror or other methods should still retain their bat form on the basis that they would likely be the only vampires and they are a non-round start race.

vampires outside of halloween are severely breaking our theme

**Comment by D4C-420** [comment]
> vampires outside of halloween are severely breaking our theme

All the Halloween races can be found in game outside of Halloween. 
zombies via romeral
cloth golems by any golem making means
dulluahns by the same way as vampires (ie pride mirror)
skeletons in quite a few ways

Besides that avoiding my point of this affecting vampires during times when they are not a round start race and it takes much more effort to become them.

**Comment by MrDoomBringer** [comment]
i was hoping this would be something more funny, say vampires not being able to enter departments without being invited in

**Comment by SuperNovaa41** [comment]
> vampires outside of halloween are severely breaking our theme

Our theme of what? Scifi?
That would mean cult, heretic, wizard, nightmere, and probably others are out of theme too

**Comment by itseasytosee** [comment]
Nah vamps are just lame.

**Comment by Orchian** [comment]
Good stuff

File: code/modules/client/preferences/species_features/vampire.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/datum/preference/choiced/vampire_status
```
Comment (Mothblocks) [inline code/modules/client/preferences/species_features/vampire.dm:1]: Sure you don't want a default value on this?

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 2]
```diff
@@ -18,51 +33,78 @@
 	mutanttongue = /obj/item/organ/tongue/vampire
 	limbs_id = "human"
 	skinned_type = /obj/item/stack/sheet/animalhide/human
+	///some starter text sent to the vampire initially, because vampires have shit to do to stay alive
 	var/info_text = "You are a <span class='danger'>Vampire</span>. You will slowly but constantly lose blood if outside of a coffin. If inside a coffin, you will slowly heal. You may gain more blood by grabbing a live victim and using your drain ability."
-	var/obj/effect/proc_holder/spell/targeted/shapeshift/bat/batform //attached to the datum itself to avoid cloning memes, and other duplicates
+	///static list shared among all vampire species datums that give a house name for each department
+	var/static/list/vampire_houses = list(
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm:39]: I don't understand. Why does this start out filled at all? It looks like it's a mixed list, which is a bad sign.

Comment (tralezab) [reply code/modules/mob/living/carbon/human/species_types/vampire.dm:39]: good point, it could be empty

File: code/modules/client/preferences/species_features/vampire.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/client/preferences/species_features/vampire.dm:1]: Sure but unsure on which is better

probably inoculated, simply to get everyone to at least try out the gimmick

Comment (Mothblocks) [reply code/modules/client/preferences/species_features/vampire.dm:1]: That was my thought, it's probably going to be fairly tougher to have enough vampires on after this

**Comment by Spookuni** [comment]
I agree with the assessment that vampires are broken bullshit when they're selectable as a roundstart race, they're decently strong for an advanced Race and when you can spawn in as them they've got no in-round limit on how often you can fuck around with their abilities and make life extremely difficult for everyone else.

That said, 98% (97.9 actually I believe, but eh.) of the time vampires are available to players in round through three sources, Admins (generally through events or antag TC trades), Pride's mirror, and the Wizard mirror. All three of these methods are unreliable and rare, and all three also allow (in theory, admin whims are mercurial) access to the full list of available advanced races, and compared to many of those other races, vampires really don't stand out as much powerwise, while filling an excellent and extremely fun niche for TC trading antagonists or people having fun with pride's mirror, a niche which I don't think it benefits the game to destroy in the name of smoothing over one week of seasonal event, vampires are annoying sometimes, but april fools emergency meetings they are not.

Powerwise, Vampires currently have associated with themselves: No_breath (Generally an upside, though conditionally a pretty major downside when atmos decides to use some of their gases for useful things) Virus immunity (Generally a downside, I see far more cookie cutter healing viruses distributed than major death viruses, and well set up healing viruses are extremely effective, losing them hurts. Plus bat form would defacto make you immune to the occasional murder-death-kill virus popping up anyway by turning you into a simplemob, and you don't breath so you'd have a harder time catching bad viruses over pill form good viruses anyway, all around it's a downside.) Blood sucking, (Shares many game play elements with ethereals honestly, you can actually die from not sucking blood unlike almost everyone else with food, but with some minor effort (ask botany for a monkey cube or make one yourself with public botany) you can stock up hard enough that when everyone else is starting to accrue foodmood you're still on 800/2000, and medical is generally completely fine with letting you borrow an IV for two minutes in the worst case anyway (In my experience, people who complain they need blood to live and randomly attack others as a non-antag are just griefers looking for an excuse to grief.) Blood sucking is overall a net positive on power in my experience) aaaaaaand batform. (Lets you turn into a 10hp simplemob bat at will with a variety of non-combat and combat effects associated. Non-combat wise, the bat can ventcrawl, and is immune to cold and low pressure, but moves slower than humans and is NOT immune to heat and high pressure, which tends to kill you extremely quickly when you have ten health. Combat wise, batform can be cast while slowed due to damage, slowed due to stamina loss, in stamina crit, cuffed, aggro, neck, or kill grabbed and legcuffed, this is undeniably extremely strong in combat where stuns are king, but is not at all an instant win button. Vampire bats have, again, only Ten health, and so die instantly to uh, basically everything, you're locked into the transformation for ~5 seconds at minimum after transforming (presumably in front of someone who is now looking to kill your 10 health having self) and while you can time out stamina crit itself in bat form, you'll revert with any knockdowns still running, any cuffs or legcuffs you were suffering from still present. And while yes you can ventcrawl, your literal only defence against immediate death as a bat is evasive action, standing still for the several seconds it takes to ventcrawl while someone is trying to kill you is a synonym for becoming a dead vampire extremely quickly. Batform also has the side effects of entirely purging all chemicals in your system - both positive and negative on transformation, as well as removing all wounds you are currently suffering from, it will also transmute all brute and burn damage you're currently suffering on any limb into brute damage specifically and on your torso, which can be an upside if you only have brute meds or sutures, but a downside if you have a mix of medical options, especially if you're a seccie using quadsec for it's perlimb heal. and ALL batform interactions have the massive asterix that you have TEN HEALTH for the duration of the time it takes to transform into batform and back again at an absolute minimum, bats are very easy to kill (and they'll also aggro any simplemobs with an attack damage near instantly, I'm not sure why, but it's incredibly easy to get bitchslapped by a non-hostile simplemob that aggroed onto your batform from off screen and sprinted over to end your fucking life))

For the easiest reference against this, the three most commonly sets of available advanced races to the station are:
1. Androids, coming from the associated chaplain sect, which grant significant armour to all types, no_breath (as with vampires), true spacewalking (as opposed to vampire's highly conditional spacewalking), increased availability of self-healing options (with a corresponding decrease in medical's efficiency), complete immunity to viruses (as with vampires), immunity to chemicals of both types (as opposed to vampire's ability to self-purge by risking switching to batform) and a crippling weakness to EMPs.
2. Podpeople, grown from botany, who never have to care about eating as long as they remain in light, with a corresponding easier time keeping their mood positive for the extra effective health and action speed that grants, plus constant healing at all times in light. Offset by a crippling weakness to fire, (and also they lose their bonii in darkness, but torches are very easy to get a hold of)
3. The slime subtypes, Though I'm going to focus specifically on slimepeople, because while the others are very powerful, they're also a fair bit more situational than undying infinite respawns. Slimepeople are toxin lovers (immune to and heal from conventional toxins, at the cost that medical will attempt to murder you with the nearest toxin treatment option the moment they lay eyes on you), capable of self-regeneration (autotomy go brrr) and have the ability to split themselves to create new bodies to do whatever with ( effectively becoming defacto immortal if they lean into it at all), their balancing crippling weakness is that they're instantly mutated by slime type mutation toxins with no ability to respond once injected, but considering their source, they generally effectively control the source of their own largest weakness.

All three of these advanced races grant intrinsic upsides comparable to the power of vampires when factoring in that none of them need to risk being instantly one shot to use their primary powers (especially now that shots by guns on combat mode automatically hit things on their tiles now), and all of them without a single question utterly eclipse vampires minus batform unless specifically countered by their primary downside, compared to vampires which are countered by an angry assistant with a welder (15 damage is more than 10 health).

I could list some of the other ADV races available by the same methods as vampires (or more, in most cases), but I think the point is made that vampires aren't really out of line powerwise when you can't spawn in as them roundstart (and honestly, I'd be more afraid of roundstart slimepeople than roundstart vampires, at least you can kill vampires.) and that's secondary to whether they're actually fun or not, which I think they very much are. They're a race with great flavour as mini-shapeshifters and even just wandering around a bat can be great RP material and very fun to work with. If Halloween is really dragged down far enough by annoying griefer vampires that it's a code issue that must be annihilated, (personally, I'm not sure it is, but different experiences from different people) fine. But please don't functionally delete an extremely fun to use advanced race that's uncommonly / rarely available and rarer still actually obtained in the name of balancing 2% of our gametime better.

TL:DR ("Spook why the fuck did you write 1400 words on stupid space game funny shapeshifter race")
Vampires roundstart during halloween: too strong, good faith in hands of players, understand concerns
Vampires outside of Halloween: Fun and rare mini-shapeshifter race, largely in line with other advanced races as difficult or easier to acquire
Please don't kill the second in the name of the problems of the first.

P.S "vampires outside of halloween are severely breaking our theme"
Others have said this, but we already have wizards and cursed items and literal magic how is a magical cursed shapeshifter species out of theme, especially when the primary way of ever playing as them is literally touching a cursed object.

**Comment by tralezab** [comment]
pride mirror is literally an admin mirror placed as lavaland loot dont even @ me

**Comment by tralezab** [comment]
@Spookuni 
bro, i... come on now

**Comment by optimumtact** [comment]
I dont' care if it's available or not, im getting rid of that stupid batform

**Comment by wesoda25** [comment]
> You can’t get skeleton or zombie normally (maybe pride?), and romerol zombies are not Halloween ones

So it wouldn't exactly be out of form to have the roundstart vamps be different from naturally attainable ones, right? 

Also arm come on you aren't even countering what they said, you are currently expanding on the thematic aspect of vamps.

**Comment by tralezab** [comment]
> you are currently expanding on the thematic aspect of vamps.

as a holiday race, if i'm bringing it out of the holiday i'm changing it to be an antagonist modifier you can spawn alongside as a traitor with some form of space rabies.

which i may still do so watch out!

**Comment by tralezab** [comment]
which i should add should not be done via species in the first place

**Comment by tralezab** [comment]
alllmost ready

**Comment by KittyNoodle** [comment]
Dude just make vamps not a Halloween race

**Comment by EricZilla** [comment]
Conceptually, could you not have the become vampire part of the species grant the spell while roundstart doesnt have it?

**Comment by tralezab** [comment]
this is officially ready

**Comment by Mothblocks** [comment]
those tests dont looko ready to me, waht hapened

**Review by Twaticus — APPROVED** [review]

**Comment by tralezab** [comment]
> those tests dont looko ready to me, waht hapened

i shrunk into a tiny owned cob of corn
